### PR TITLE
Tarrif service

### PR DESCRIPTION
**Pull request by AGradowski**

### CODE REVIEW

File: contracts/test/test_rental.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      assert.equal(result[0][1],2);
+      assert.equal(result[0][2],3);
+      //availability
+      assert.equal(result[1][0],true);
+      assert.equal(result[1][1],false);
+      assert.equal(result[1][2],false);
+      //assert.equal(result[1],2);  
+
+  });
+
```
Comment (JanuszJakubiec) [inline contracts/test/test_rental.js:88]:
```suggestion
```

File: contracts/test/test_rental.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
+const Rental = artifacts.require("../contracts/Rental.sol");
+const chai = require('chai')
+const chaiAsPromised = require('chai-as-promised');
+
+
```
Comment (JanuszJakubiec) [inline contracts/test/test_rental.js:5]: Why are there that many empty lines? Maybe reduce to single empty line.

File: contracts/test/test_rental.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      return this.rentalTest.addCar(1).then(() => {
+        assert.ok(false, "It didn't fail");
+      }, () => {
+        assert.ok(true, "Passed");
+      });
+    });
+
+    it("should add active rental and get it", async () => {
+        const emit_event = await this.rentalTest.startRental(10,1,1);
+
```
Comment (JanuszJakubiec) [inline contracts/test/test_rental.js:41]: Wy are there so many empty lines?

File: contracts/test/test_rental.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      const result = await this.rentalTest.getAllAvailableCars();
+      //cars
+      assert.equal(result[0][0],1);
+      assert.equal(result[0][1],2);
+      assert.equal(result[0][2],3);
+      //availability
+      assert.equal(result[1][0],true);
+      assert.equal(result[1][1],false);
+      assert.equal(result[1][2],false);
+      //assert.equal(result[1],2);
```
Comment (JanuszJakubiec) [inline contracts/test/test_rental.js:85]: why is this line commented out?

File: contracts/test/test_rental.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      //cars
+      assert.equal(result[0][0],1);
+      assert.equal(result[0][1],2);
+      assert.equal(result[0][2],3);
+      //availability
+      assert.equal(result[1][0],true);
+      assert.equal(result[1][1],false);
+      assert.equal(result[1][2],false);
+      //assert.equal(result[1],2);  
+
```
Comment (JanuszJakubiec) [inline contracts/test/test_rental.js:86]: Why is this line empty?

File: contracts/test/test_tarrif.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    assert.equal(result[1][0], 10);
+
+    assert.equal(result[0][1], 2);
+    assert.equal(result[1][1], 20);
+
+    assert.equal(result[0][2], 3);
+    assert.equal(result[1][2], 30);
+
+  });
+});
```
Comment (JanuszJakubiec) [inline contracts/test/test_tarrif.js:40]: Please add newline at the end of file

File: contracts/test/test_tarrif.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+    assert.equal(result[0][0], 1);
+    assert.equal(result[1][0], 10);
+  });
+
+  it('should add complex pricing and return them', async()=>{
+   //entry (1,10) from previous test
+    await this.tarrifTest.addEntry(2,20);
+    await this.tarrifTest.addEntry(3,30);
+    const result =await  this.tarrifTest.getPricing();
```
Comment (JanuszJakubiec) [inline contracts/test/test_tarrif.js:28]:
```suggestion
    const result = await  this.tarrifTest.getPricing();
```

File: contracts/test/test_tarrif.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    await this.tarrifTest.addEntry(1,10);
+    const result =await  this.tarrifTest.getPricing();
+
+    assert.equal(result[0][0], 1);
+    assert.equal(result[1][0], 10);
+  });
+
+  it('should add complex pricing and return them', async()=>{
+   //entry (1,10) from previous test
+    await this.tarrifTest.addEntry(2,20);
```
Comment (JanuszJakubiec) [inline contracts/test/test_tarrif.js:26]: please add space after ","

File: contracts/test/test_rental.js [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+      });
+    });
+
+    it("should return two cars (array of three, with two marked as free (so false)), after one has been rented", async () => {
+      await this.rentalTest.addCar(3);
+      await this.rentalTest.startRental(10,1,1);
+
+      const result = await this.rentalTest.getAllAvailableCars();
+      //cars
+      assert.equal(result[0][0],1);
```
Comment (JanuszJakubiec) [inline contracts/test/test_rental.js:78]: please add space after ","

File: contracts/test/test_tarrif.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+  it('should deploy successfully', async () => {
+    assert.notEqual(creatorAddress, 0x0);
+    assert.notEqual(creatorAddress, '');
+    assert.notEqual(creatorAddress, null);
+    assert.notEqual(creatorAddress, undefined);
+  });
+
+  it('should add one pricing and return it', async()=>{
+    await this.tarrifTest.addEntry(1,10);
```
Comment (JanuszJakubiec) [inline contracts/test/test_tarrif.js:17]: please add space after ","

File: tarrif/build.gradle [hunk 11]
```diff
@@ -0,0 +1,61 @@
+plugins {
+    id 'org.springframework.boot' version '2.6.5' apply false
```
Comment (JanuszJakubiec) [inline tarrif/build.gradle:2]:
```suggestion
    id 'org.springframework.boot' version '2.6.5'
```

Comment (JanuszJakubiec) [reply tarrif/build.gradle:2]: If possible, delete "apply false"

File: tarrif/config/config.json [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+    "auth": "5011",
+    "rental": "5021"
+  },
+  "addresses": {
+    "blockchain": "0.0.0.0",
+    "tarrif": "127.0.0.1",
+    "auth": "127.0.0.1",
+    "rental": "127.0.0.1"
+  }
+}
```
Comment (JanuszJakubiec) [inline tarrif/config/config.json:14]: Please add new line at the end of file

File: tarrif/src/main/java/pl/edu/agh/tarrif/errors/ErrorResponseDto.java [hunk 13]
```diff
@@ -0,0 +1,5 @@
+package pl.edu.agh.tarrif.errors;
+
+public record ErrorResponseDto(
+        String message
```
Comment (JanuszJakubiec) [inline tarrif/src/main/java/pl/edu/agh/tarrif/errors/ErrorResponseDto.java:4]: Why is it split between 3 lines?

File: tarrif/src/main/java/pl/edu/agh/tarrif/tarrif/TarrifController.java [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+public class TarrifController {
+    private final TarrifService tarrifService;
+
+    public TarrifController(final TarrifService tarrifService) {
+        this.tarrifService = tarrifService;
+    }
+
+    @GetMapping("/pricing")
+    @ResponseStatus(HttpStatus.OK)
+    public ArrayList<PricingRecord> getPricing(@CurrentUser final User user){
```
Comment (JanuszJakubiec) [inline tarrif/src/main/java/pl/edu/agh/tarrif/tarrif/TarrifController.java:24]: If you are not using user in a function in controller, you don't have to pass it in an argument. Spring inserts the user if you include annotation CurrentUser. I suggest deleting this if possible.

File: tarrif/src/main/java/pl/edu/agh/tarrif/tarrif/service/TarrifService.java [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+            System.out.println(e);
+            return false;
+        }
+    }
+
+    public ArrayList<PricingRecord> getPricing(){
+        ArrayList<PricingRecord> pricing = new ArrayList<>();
+        try {
+            Tuple3<List<BigInteger>, List<BigInteger>,List<BigInteger>> res=tarrifBlockchainProxy.getPricing();
+            for(int i =0; i<res.component1().size();i++){
```
Comment (JanuszJakubiec) [inline tarrif/src/main/java/pl/edu/agh/tarrif/tarrif/service/TarrifService.java:37]:
```suggestion
            for(int i =0; i<res.component1().size(); i++){
```

File: tarrif/src/main/java/pl/edu/agh/tarrif/tarrif/service/TarrifService.java [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+        } catch (Exception e) {
+            System.out.println(e);
+            return false;
+        }
+    }
+
+    public ArrayList<PricingRecord> getPricing(){
+        ArrayList<PricingRecord> pricing = new ArrayList<>();
+        try {
+            Tuple3<List<BigInteger>, List<BigInteger>,List<BigInteger>> res=tarrifBlockchainProxy.getPricing();
```
Comment (JanuszJakubiec) [inline tarrif/src/main/java/pl/edu/agh/tarrif/tarrif/service/TarrifService.java:36]:
```suggestion
            Tuple3<List<BigInteger>, List<BigInteger>, List<BigInteger>> res = tarrifBlockchainProxy.getPricing();
```

File: tarrif/src/main/java/pl/edu/agh/tarrif/tarrifContract/Tarrif.java [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
...5061040c565b5085935050505092915050565b6000606082019050818103600083015261046081866103e8565b9050818103602083015261047481856103e8565b9050818103604083015261048881846103e8565b9050949350505050565b600080fd5b6104a0816103aa565b81146104ab57600080fd5b50565b6000813590506104bd81610497565b92915050565b600080604083850312156104da576104d9610492565b5b60006104e8858286016104ae565b92505060206104f9858286016104ae565b9150509250929050565b7f4e487b7100000000000000000000000000000000000000000000000000000000600052604160045260246000fd5b7f4e487b7100000000000000000000000000000000000000000000000000000000600052603260045260246000fd5b7f4e487b7100000000000000000000000000000000000000000000000000000000600052601160045260246000fd5b600061059b826103aa565b91507fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff82036105cd576105cc610561565b5b600182019050919050565b600082825260208201905092915050565b7f41636365732064656e696564202d206e6f2061646d696e206163636573730000600082015250565b600061061f601e836105d8565b915061062a826105e9565b602082019050919050565b6000602082019050818103600083015261064e81610612565b905091905056fea26469706673582212208ed516d543c0d66529540893814aeb5c132688baa2f96352bfc4a4195b2a1b0164736f6c634300080e0033";
```
Comment (JanuszJakubiec) [inline tarrif/src/main/java/pl/edu/agh/tarrif/tarrifContract/Tarrif.java:35]: I'm not completely against that infinitely long line, but If possible try to change it to finate long line.

File: tarrif/src/test/java/TarrifServiceTest.java [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+    private final List<BigInteger> intList1 = Lists.newArrayList(intArray);
+    private final List<BigInteger> intList2 = Lists.newArrayList(intArray);
+    private final List<BigInteger> intList3 = Lists.newArrayList(intArray);
+    @Test
+    @Transactional
+    public void testAddPricing() throws Exception {
+        when(tarrifBlockchainProxy.addEntry(any())).thenReturn(null);
+        this.tarrifService = new TarrifService(tarrifBlockchainProxy);
+        try{
+            assertThat(tarrifService.addPricing(new PricingRecord(1,100))).isEqualTo(true);
```
Comment (JanuszJakubiec) [inline tarrif/src/test/java/TarrifServiceTest.java:45]:
```suggestion
            assertThat(tarrifService.addPricing(new PricingRecord(1, 100))).isEqualTo(true);
```

**Review by JanuszJakubiec — CHANGES_REQUESTED** [review]
Looks good, I have added some comments regarding formatting issues.

File: contracts/test/test_rental.js (hunk 3 above)
Comment (AGradowski) [reply contracts/test/test_rental.js:41]: For clarity, but true, there were too many. Changed it.

File: contracts/test/test_rental.js (hunk 4 above)
Comment (AGradowski) [reply contracts/test/test_rental.js:85]: Left from testing the tests, I removed it now (it was wrong anyway)

File: contracts/test/test_rental.js (hunk 1 above)
Comment (AGradowski) [reply contracts/test/test_rental.js:88]: Fixed

File: contracts/test/test_rental.js (hunk 2 above)
Comment (AGradowski) [reply contracts/test/test_rental.js:5]: Fixed

File: contracts/test/test_rental.js (hunk 5 above)
Comment (AGradowski) [reply contracts/test/test_rental.js:86]: Fixed

File: contracts/test/test_tarrif.js (hunk 6 above)
Comment (AGradowski) [reply contracts/test/test_tarrif.js:40]: Done

File: contracts/test/test_tarrif.js (hunk 7 above)
Comment (AGradowski) [reply contracts/test/test_tarrif.js:28]: Fixed

File: contracts/test/test_tarrif.js (hunk 8 above)
Comment (AGradowski) [reply contracts/test/test_tarrif.js:26]: Done

File: contracts/test/test_rental.js (hunk 9 above)
Comment (AGradowski) [reply contracts/test/test_rental.js:78]: Done

File: contracts/test/test_tarrif.js (hunk 10 above)
Comment (AGradowski) [reply contracts/test/test_tarrif.js:17]: Done

File: tarrif/build.gradle (hunk 11 above)
Comment (AGradowski) [reply tarrif/build.gradle:2]: Done

File: tarrif/config/config.json (hunk 12 above)
Comment (AGradowski) [reply tarrif/config/config.json:14]: Done

File: tarrif/src/main/java/pl/edu/agh/tarrif/errors/ErrorResponseDto.java (hunk 13 above)
Comment (AGradowski) [reply tarrif/src/main/java/pl/edu/agh/tarrif/errors/ErrorResponseDto.java:4]: Fixed - now is in one line

File: tarrif/src/main/java/pl/edu/agh/tarrif/tarrif/TarrifController.java (hunk 14 above)
Comment (AGradowski) [reply tarrif/src/main/java/pl/edu/agh/tarrif/tarrif/TarrifController.java:24]: Removed the user argument, as it was not used

File: tarrif/src/main/java/pl/edu/agh/tarrif/tarrif/service/TarrifService.java (hunk 15 above)
Comment (AGradowski) [reply tarrif/src/main/java/pl/edu/agh/tarrif/tarrif/service/TarrifService.java:37]: Done

File: tarrif/src/main/java/pl/edu/agh/tarrif/tarrif/service/TarrifService.java (hunk 16 above)
Comment (AGradowski) [reply tarrif/src/main/java/pl/edu/agh/tarrif/tarrif/service/TarrifService.java:36]: Done

File: tarrif/src/test/java/TarrifServiceTest.java (hunk 18 above)
Comment (AGradowski) [reply tarrif/src/test/java/TarrifServiceTest.java:45]: Done

File: contracts/test/test_tarrif.js (hunk 7 above)
Comment (JanuszJakubiec) [reply contracts/test/test_tarrif.js:28]: It was not fixed

Comment (JanuszJakubiec) [reply contracts/test/test_tarrif.js:28]: You forgot to fix this

**Review by JanuszJakubiec — CHANGES_REQUESTED** [review]
One line is left uncorrected. The rest looks good.

**Review by JanuszJakubiec — APPROVED** [review]

**Pull request merged**